xarray/testing.py

```diff
@@ -0,0 +1,95 @@
+"""Testing functions exposed to the user API"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+from xarray.core import ops
+
+
+def decode_string_data(data):
+    if data.dtype.kind == 'S':
+        return np.core.defchararray.decode(data, 'utf-8', 'replace')
+    return data
+
+
+def data_allclose_or_equiv(arr1, arr2, rtol=1e-05, atol=1e-08):
+    if any(arr.dtype.kind == 'S' for arr in [arr1, arr2]):
+        arr1 = decode_string_data(arr1)
+        arr2 = decode_string_data(arr2)
+    exact_dtypes = ['M', 'm', 'O', 'U']
+    if any(arr.dtype.kind in exact_dtypes for arr in [arr1, arr2]):
+        return ops.array_equiv(arr1, arr2)
+    else:
+        return ops.allclose_or_equiv(arr1, arr2, rtol=rtol, atol=atol)
+
+
+def assert_dataset_allclose(d1, d2, rtol=1e-05, atol=1e-08):
```

Comment:
Can we delete this method instead? I think `assert_allclose` covers its functionality.